packages/react-test-renderer/src/__tests__/ReactShallowRenderer-test.js

```diff
@@ -915,6 +915,39 @@ describe('ReactShallowRenderer', () => {
     expect(result.props.children).toEqual(3);
   });
 
+  it('should call getDerivedStateFromProps with the state set by setState', () => {
+    class SimpleComponent extends React.Component {
```

Comment:
Is this based on any existing test in other renderers? If not, can you please find an appropriate existing test and then use it as a base?

Reply:
I'll look for that!

Reply:
The only test I could find that was related to this was: "should not override state with stale values if prevState is spread within getDerivedStateFromProps" in `react/packages/react-dom/src/__tests__/ReactComponentLifeCycle-test.js`. I modified it a bit (it involved a parent and a child component, but in the case of shallow rendering this seemed to be a bit too complicated).